airflow/jobs/base_job.py

```diff
@@ -171,17 +171,14 @@ def heartbeat(self):
             if self.state == State.SHUTDOWN:
                 self.kill()
 
-            is_unit_test = conf.getboolean('core', 'unit_test_mode')
```

Comment:
Do you mind also include the reason why do we want to remove the `is_unit_test` check  please?

Reply:
This if condition was added to speed up the tests, however, there is a much better and more maintainable way to achieve the same effect, which is mock the sleep call from unit tests. By doing that, we get the following benefits:

1. in unit test, we can now check and assert whether sleep has been called with the right argument
2. avoid unnecessary dictionary lookup and if statement check at runtime
3. avoid the anti-pattern where unit test logic is mingled with business logic. unit test should be as transparent to the actual code as possible

I noticed we are adopting this pattern in 2 other places as well and I really think we should replace all of them with proper mocking in the test and keep the business code clean.

Reply:
I agree in principle @houqp. But should not we remove the check and add mocking it in the same commit? I vaguely remember that this single "if" caused few minutes shorter CI build - so if we remove it now, everyone will be impacted.

Reply:
@potiuk I have already added the mock in this PR. Or are you thinking of separating out the mock change into its own commit?

Reply:
I am thinking of patching sleep globally for all unit tests, so we won't have to worry about it going forward. is this the kind of change you all comfortable with?

Reply:
@KevinYang21 @potiuk is there anything you want me to change>

Reply:
> I am thinking of patching sleep globally for all unit tests, so we won't have to worry about it going forward. is this the kind of change you all comfortable with?

No, sometimes we _need_ sleep in our tests to give a sub-process time to do start up etc.(mostly tests, not in the airflow core), or in the dags to "do nothing for a long time" so this would break all sorts of things.

Reply:
That's fair, I will leave it to test level like it is right now then.

Reply:
@KevinYang21 @potiuk @ashb are you cool with the unit test changes I made in this PR? If not, please feel free to comment. This is a pretty annoying bug and renders all task logs not readable.